web/modules/custom/pipeline/src/Controller/PipelineExecutionController.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\pipeline\Controller;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Url;
+use Drupal\pipeline\PipelineOrchestratorInterface;
+use Drupal\pipeline\PipelineStateManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * Defines the pipeline execution controller.
+ */
+class PipelineExecutionController extends ControllerBase {
+
+  /**
+   * The pipeline orchestrator service.
+   *
+   * @var \Drupal\pipeline\PipelineOrchestratorInterface
+   */
+  protected $orchestrator;
+
+  /**
+   * The persistent state of the importer.
```

Comment:
This is not joinup specific any more so this is not an importer. Maybe "The state manager service" simply.